tests/e2e/test_jenkins.py

```diff
@@ -16,13 +16,20 @@
 import sys
 
 from cibyl.cli.main import main
-from tests.e2e.fixture import JenkinsTest
+from tests.e2e.fixtures import JenkinsTest
 
 
 class TestJenkins(JenkinsTest):
     """Tests queries regarding the Jenkins source.
     """
 
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.add_job('test_1')
```

Comment:
I think it might be worth to wrap this calls in a try/except block and call the tearDownClass method in the except before reraising the exception. Otherwise in case of failure the container is not stopped and I get an error when trying to create the gerrit container for zuul because thay both use the same port

Reply:
Good point, I will see if I can make that happen.

Reply:
Investigating a little, I have found that using 'with' will be much safer than 'tearDown'. The 'with' keyword guarantees that no matter reason the container will be closed. The pull requests grows bigger in scope with the latest changes but I think it will be worth it.